### PR TITLE
Modified PINNBenchmark 

### PR DESCRIPTION
**Pull request by iamyixuan**
Added RunningJob, profile decorator, LCModelStopper, metadata, and README.

### CODE REVIEW

File: lib/PINNBench/Burgers/README.md
Comment (Deathn0t) [inline lib/PINNBench/Burgers/README.md:1]: Could you add some more details in the README:
- What metrics are optimised?
- What is the max-min budget?
- What is the unit of this budget (epochs?)?

File: lib/PINNBench/Burgers/benchmark.py [hunk 1]
```diff
@@ -11,5 +10,8 @@ class PINNBurgersBenchmark(Benchmark):
     version = "0.0.1"
 
     requires = {
-        "py-pip-requirements": {"type": "pip", "name": "-r "+ os.path.join(DIR, "requirements.txt")}, # not sure what this line should do or I should do to match this line.
+        "py-pip-requirements": {
+            "type": "pip",
+            "name": "-r " + os.path.join(DIR, "requirements.txt"),
+        },  # not sure what this line should do or I should do to match this line.
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/benchmark.py:16]: `# not sure what this line should do or I should do to match this line.` maybe this line can be removed

File: lib/PINNBench/Burgers/hpo.py [hunk 2]
```diff
@@ -1,54 +1,104 @@
...
-# define the run function
-def run(config):
-    nu=0.01/np.pi
-    train, val, test = get_data(NT=200, NX=128, X_U=1, X_L=-1, T_max=1,
-                                Nu=nu)
     num_layers = config["num_layers"]
     hidden_dim = config["hidden_dim"]
     output_dim = 1
-    epochs = config["epochs"]
+    epochs = 1  #
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/hpo.py:23]: What does this parameter change?

File: lib/PINNBench/Burgers/hpo.py [hunk 3]
```diff
@@ -1,54 +1,104 @@
...
+    for budget_i in range(min_b, max_b + 1):
+        eval = sup.step(train, val)
+        print("val mse", eval)
+        objective_i = -eval  # maximizing in deephyper
+        job.record(budget_i, objective_i)
+        if job.stopped():
+            break
+    objective = job.objective
+
+    metadata = {
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/hpo.py:44]: Is it possible to return other metadata?
- train/val/test loss
- train/val/test metrics
- number of trainable parameters of the model?
I started making a list of the "generic" metadata that all benchmarks could return here: https://github.com/deephyper/benchmark/issues/4

File: lib/PINNBench/Burgers/hpo.py [hunk 4]
```diff
@@ -1,54 +1,104 @@
...
+        hidden_dim=hidden_dim,
+        output_dim=output_dim,
+        act_fn=activation,
+    )
     sup = BurgerSupervisor(nu=nu, net=net, epochs=epochs, lr=lr, alpha=alpha)
-    val_f_loss = sup.train(train, test) # evaluate the best config on the testing set.
+    val_f_loss = sup.train(train, test)  # evaluate the best config on the testing set.
+
+
+if __name__ == "__main__":
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/hpo.py:91]: Maybe we should have some `evaluate_baseline` baseline function which would load the default problem configuration and the `evaluate` function to train/test the baseline configuration. We should also probably report this performance in the README.

File: lib/PINNBench/Burgers/requirements.txt [hunk 5]
```diff
@@ -1,6 +1,5 @@
-deephyper==0.4.2
+deephyper==0.5.0
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/requirements.txt:1]: I suggest replacing with `deephyper>=0.5.0` to be less restrictive. It is less safe that's true... so maybe we should have some `install(..., strict=True/False)` to be able to install the "strict" requirements (tested) and something a bit softer.

File: lib/PINNBench/Burgers/hpo.py (hunk 2 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/hpo.py:23]: With the `LCModelStopper`, the number of epoch for each run is set to 1.

Comment (Deathn0t) [reply lib/PINNBench/Burgers/hpo.py:23]: But isn't the `LCModelStopper` deciding what number of epochs to run?
